app/controllers/concerns/secure_headers_concern.rb

```diff
@@ -5,11 +5,16 @@ module SecureHeadersConcern
 
   def apply_secure_headers_override
     return if stored_url_for_user.blank?
-    return unless IdentityConfig.store.openid_connect_content_security_form_action_enabled
 
     authorize_form = OpenidConnectAuthorizeForm.new(authorize_params)
     return unless authorize_form.valid?
 
+    return if !IdentityConfig.store.openid_connect_content_security_form_action_enabled &&
+              oidc_redirect_method(
+                issuer: authorize_form.service_provider.issuer,
+                user_uuid: current_user&.uuid,
+              ) != 'server_side'
```

Comment:
this is a really long line and it's repeated, what if we made an named helper for this (with a better name than I am proposing)

```suggestion
    return if csp_disabled_and_not_server_side?
```

Reply:
i like this suggestion, and honestly i think that name is fine? it's more clear what's happening

Reply:
yep!

Reply:
Added a method and shuffled things a bit into a concern in 3ee3e6c